pkg/virt-controller/watch/vm_test.go

```diff
@@ -5492,6 +5492,68 @@ var _ = Describe("VirtualMachine", func() {
 					Expect(err).To(Succeed())
 					Expect(vm.Spec.Preference.RevisionName).To(Equal(expectedPreferenceRevision.Name))
 				})
+
+				It("should apply preferences to hot plugged interface in VMI", func() {
```

Comment:
In continuation to the concern raised in the production code feedback, it may be useful to cover these scenarios as well:
- An interface is plugged but at the same time another is unplugged (and targeted for being removed completely from the spec). Note that an unplug flow has two steps: 1) mark the interface as `absent` and 2) when all conditions are met, remove the record completely.
- Simulate that changes have been done to the VM that are not hotplug (i.e. passed to the VMI) which may effect the preferences (e.g. removing or adding a field value from the spec that may now trigger the preference).

Reply:
> * An interface is plugged but at the same time another is unplugged (and targeted for being removed completely from the spec). Note that an unplug flow has two steps: 1) mark the interface as `absent` and 2) when all conditions are met, remove the record completely.

I can add a test case but my assumption was that the current conditional would still catch the new interface while the interface being removed is still present and marked as `absent`.

> * Simulate that changes have been done to the VM that are not hotplug (i.e. passed to the VMI) which may effect the preferences (e.g. removing or adding a field value from the spec that may now trigger the preference).

I think this generic preference behavioural test falls under https://github.com/kubevirt/kubevirt/issues/10175 but I can add something in a follow up after this.

Reply:
> I can add a test case but my assumption was that the current conditional would still catch the new interface while the interface being removed is still present and marked as `absent`.

That covers the scenario where the interface is present and marked for unplug.
But there is another scenario where the interface is in practice removed completely (it will no longer appear on the interface/network list).

In practice, the condition `if len(updatedVmiSpec.Domain.Devices.Interfaces) > len(vmiCopy.Spec.Domain.Devices.Interfaces)` is currently assuming that `vmiCopy` has these interfaces already removed. That is indeed the case, but this can also change after a code refactor and then nothing will validate it broke something.
These tests explain it well: https://github.com/kubevirt/kubevirt/blob/6ece9009ee8b227b7eb8410730c5c71c0fc98cd6/pkg/virt-controller/network/hotplug_test.go#L268